front-end/stateofthecards-front-end/src/components/QuickJoinMenu.tsx

```diff
@@ -0,0 +1,144 @@
+import React from "react";
+import styles from "./QuickJoinMenu.module.css";
+import ScaleLoader from "react-spinners/ScaleLoader";
+
+enum QuickJoinState {
+	Idle,
+	Searching,
+	InLobby,
+}
+
+interface IProps {
+	onClickCancel: () => void;
+}
+
+interface IState {
+	currentState: QuickJoinState;
+	lobbyInfo?: LobbyInfo;
+}
+
+interface LobbyInfo {
+	// The current players connected to the lobby
+	//  Should not use string-type in the future.
+	//  so that a player can press on a different,
+	//  player to show info.
+	players: string[];
+
+	// The id of the game so it can
+	//  be fetched by the user.
+	gameInfo: GameInfo;
+
+	// Should be an enum similar to:
+	//  Waiting, Preparing, Post
+	state: string;
+}
+
+// Temporary interface that contains info about
+//  the game that is going to be played.
+interface GameInfo {
+	minPlayers: number;
+	maxPlayers: number;
+	name: string;
+}
+
+class QuickJoinMenu extends React.Component<IProps, IState> {
+	constructor(props: IProps) {
+		super(props);
+
+		this.state = {
+			currentState: QuickJoinState.Idle,
+			lobbyInfo: undefined,
+		};
+	}
+
+	onClickCancel() {
+		this.props.onClickCancel();
+		this.setState({ currentState: QuickJoinState.Idle });
+	}
+
+	// Use this  method when info about a lobby has been retrieved.
+	//  Will automatically change the view to the lobby or searching view.
+	setLobbyInfo(lobbyInfo?: LobbyInfo) {
+		// Set the state depending on the given lobby-info
+		if (!lobbyInfo)
+			this.setState({ currentState: QuickJoinState.Searching });
+		else this.setState({ currentState: QuickJoinState.InLobby });
+
+		this.setState({ lobbyInfo: lobbyInfo });
+	}
+
+	render() {
+		switch (this.state.currentState) {
+			case QuickJoinState.Idle:
```

Comment:
Why is this case the only one that has it's value inline instead of in a separate function?